This is a repository for National Resource for Network Biology (NRNB) [Google Season of Docs project](https://developers.google.com/season-of-docs/docs/participants) named

# Replacing GUI control tutorials to Jupyter Notebook and R Markdown

## The work we've done
We have written notebook documents that automates the Cytoscape tutorial workflows in http://tutorials.cytoscape.org/ with R and Python.
You can check the our project progress and the links to the commits with the links below.
Our basic commit rule is _**Commit every tutorial slide**_.

### Getting Started

- [ ] ["Basic Data Visualization"](https://cytoscape.org/cytoscape-tutorials/protocols/basic-data-visualization/#/) with Python
- [x] ["Basic Data Visualization"](https://nrnb.org/gsod2019_kozo_nishida/html_documents/Rmd/basic-data-visualization.html) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/11)
- [ ] ["Differentially Expressed Genes"](https://cytoscape.org/cytoscape-tutorials/protocols/differentially-expressed-genes/#/) with Python
- [x] ["Differentially Expressed Genes"](https://nrnb.org/gsod2019_kozo_nishida/html_documents/Rmd/differentially-expressed-genes) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/7)

### Protocols

#### Cytoscape Apps

- [ ] [stringApp](https://cytoscape.github.io/cytoscape-tutorials/protocols/stringApp/#/) with Python
- [x] [stringApp](https://nrnb.org/gsod2019_kozo_nishida/html_documents/Rmd/stringApp.html) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/8)
- [ ] [WikiPathways App](https://cytoscape.github.io/cytoscape-tutorials/protocols/wikipathways-app/#/) with Python
- [x] [WikiPathways App](https://nrnb.org/gsod2019_kozo_nishida/html_documents/Rmd/wikipathways-app.html) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/10)

#### Data Visualization

- [ ] [Visualizing Data](https://cytoscape.github.io/cytoscape-tutorials/protocols/mapping-data/#/) with Python
- [ ] [Visualizing Data](https://cytoscape.github.io/cytoscape-tutorials/protocols/mapping-data/#/) with R
- [ ] [Network Layout](https://cytoscape.github.io/cytoscape-tutorials/protocols/network-layout/#/) with Python
- [ ] [Network Layout](https://cytoscape.github.io/cytoscape-tutorials/protocols/network-layout/#/) with R
- [ ] [Custom Graphics and Labels](https://cytoscape.github.io/cytoscape-tutorials/protocols/custom-enhanced-graphics-style/#/) with Python
- [x] [Custom Graphics and Labels](http://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/Custom-Graphics.nb.html) with R

#### Importing Networks and Tables

- [ ] [Loading Networks](https://cytoscape.github.io/cytoscape-tutorials/protocols/loading-networks/#/) with Python
- [ ] [Loading Networks](https://cytoscape.github.io/cytoscape-tutorials/protocols/loading-networks/#/) with R
- [ ] [Importing Data From Tables](https://cytoscape.github.io/cytoscape-tutorials/protocols/importing-data-from-tables/#/) with Python
- [x] [Importing Data From Tables](http://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/Importing-data.nb.html) with R (Already written by the mentor)
- [ ] [Importing Network From Table](https://cytoscape.org/cytoscape-tutorials/protocols/importing-network-from-table/#/) with Python
- [ ] [Importing Network From Table](https://cytoscape.org/cytoscape-tutorials/protocols/importing-network-from-table/#/) with R
- [ ] [Identifier Mapping](https://cytoscape.org/cytoscape-tutorials/protocols/identifier-mapping/#/) with Python
- [x] [Identifier Mapping](http://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/Identifier-mapping.nb.html) with R (Already written by the mentor)

#### Network Analysis

- [ ] [Differentially Expressed Genes Network Analysis](https://cytoscape.org/cytoscape-tutorials/protocols/differentially-expressed-genes/#/) with Python
- [ ] [Differentially Expressed Genes Network Analysis](https://cytoscape.org/cytoscape-tutorials/protocols/differentially-expressed-genes/#/) with R
- [ ] [Affinity Purification-Mass Spectrometry Network Analysis](https://cytoscape.org/cytoscape-tutorials/protocols/AP-MS-network-analysis/#/) with Python
- [x] [Affinity Purification-Mass Spectrometry Network Analysis](http://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/AP-MS-network-analysis.nb.html) with R (Already written by the mentor)
- [ ] [Variant Data Analysis](https://cytoscape.org/cytoscape-tutorials/protocols/variant-data-analysis/#/) with Python
- [x] [Variant Data Analysis](http://cytoscape.org/cytoscape-automation/for-scripters/R/notebooks/Cancer-networks-and-data.nb.html) with R (Already written by the mentor)
- [ ] [EnrichmentMap Pipeline](https://cytoscape.github.io/cytoscape-tutorials/protocols/enrichmentmap-pipeline/#/) with Python
- [ ] [EnrichmentMap Pipeline](https://cytoscape.github.io/cytoscape-tutorials/protocols/enrichmentmap-pipeline/#/) with R
- [ ] [Functional Enrichment](https://cytoscape.github.io/cytoscape-tutorials/protocols/functional-enrichment/#/) with Python
- [ ] [Functional Enrichment](https://cytoscape.github.io/cytoscape-tutorials/protocols/functional-enrichment/#/) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/6)
- [ ] [Filtering by Selection](https://cytoscape.github.io/cytoscape-tutorials/protocols/filtering-by-selection/#/) with Python
- [ ] [Filtering by Selection](https://cytoscape.github.io/cytoscape-tutorials/protocols/filtering-by-selection/#/) with R

### Exporting and Publishing

- [ ] [Saving Results](https://cytoscape.github.io/cytoscape-tutorials/protocols/saving-results/#/) with Python
- [x] [Saving Results](http://nrnb.org/gsod2019_kozo_nishida/html_documents/Rmd/saving-results.html) with R [links to the commits](https://github.com/nrnb/gsod2019_kozo_nishida/issues/9)

## There is still work to be done...

- We prioritize the creation of notebooks using RCy3. Python notebooks with py2cytoscape are still almost untouched.
- Python notebooks doesn't require natural language writing ability (Because the text is the same as that of RCy3), but we may need to add the missing functions in py2cytoscape.
- We will summarize what is missing in CyREST, RCy3, py2cytoscape, and stringApp commands(etc.). (The summary will be added somewhere in this repository.)

## Technical writer and mentors
### Writer
- https://github.com/kozo2
### Mentors
- https://github.com/AlexanderPico
- https://github.com/khanspers
